### PR TITLE
servermaster(engine): add check and timeout for querying job detail

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7613

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/jobmanager.go [hunk 1]
```diff
@@ -381,7 +383,7 @@ func (jm *JobManagerImpl) ListJobs(ctx context.Context, req *pb.ListJobsRequest)
 		}
 
 		// if job status is running, forward the request to jobmaster openapi
-		if canQueryJobDetail(masterMetas[i].State) {
+		if jm.canQueryJobDetail(masterMetas[i]) {
 			detail, errJob := jm.jobHTTPClient.GetJobDetail(ctx, masterMetas[i].Addr, masterMetas[i].ID)
```
Comment (CharlesCheung96) [inline engine/servermaster/jobmanager.go:387]: Maybe we should use context to control the timeout? Besides, what is the difference between `http client timeout` and `context timeout`?

Comment (gozssky) [reply engine/servermaster/jobmanager.go:387]: The `http client timeout` is applied for all requests, while `context timeout` only affects one request. There is not much difference between using context timeout or client timeout to control the request timeout. But it's easy to forget to set the context timeout in every place, which can lead to some security issues. This is why I prefer to set the client timeout.

**Review by CharlesCheung96 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**